src/Data/Biapplicative.hs

```diff
@@ -259,6 +277,119 @@ traverseBiaConst _f (Const x) = bipure (Const x) (Const x)
 traverseBiaPair :: Biapplicative p => (a -> p b c) -> (e, a) -> p (e, b) (e, c)
 traverseBiaPair f (x,y) = bimap ((,) x) ((,) x) (f y)
 
+#if MIN_VERSION_containers (0,5,8)
+{-# INLINE traverseBiaMap #-}
+traverseBiaMap :: Biapplicative p => (a -> p b c) -> Map.Map k a -> p (Map.Map k b) (Map.Map k c)
+traverseBiaMap f = \m -> go m
+  where
+    go Map.Tip = bipure Map.Tip Map.Tip
+    go (Map.Bin 1 k v _ _) =
+      bimap (\v' -> Map.Bin 1 k v' Map.Tip Map.Tip)
+            (\v' -> Map.Bin 1 k v' Map.Tip Map.Tip)
+            (f v)
+    go (Map.Bin s k v l r) =
+      biliftA3 (flip (Map.Bin s k)) (flip (Map.Bin s k))
+               (go l) (f v) (go r)
+
+{-# INLINE traverseBiaIntMap #-}
+traverseBiaIntMap :: Biapplicative p => (a -> p b c) -> IM.IntMap a -> p (IM.IntMap b) (IM.IntMap c)
+traverseBiaIntMap f = go
+  where
+    go IM.Nil = bipure IM.Nil IM.Nil
+    go (IM.Tip k v) = bimap (IM.Tip k) (IM.Tip k) (f v)
+    go (IM.Bin p m l r)
+      | m < 0     = biliftA2 (flip (IM.Bin p m)) (flip (IM.Bin p m)) (go r) (go l)
+      | otherwise = biliftA2 (IM.Bin p m) (IM.Bin p m) (go l) (go r)
+
+{-# INLINE traverseBiaTree #-}
+traverseBiaTree :: Biapplicative p => (a -> p b c) -> Tree.Tree a -> p (Tree.Tree b) (Tree.Tree c)
```

Comment:
Should `traverseBiaTree` exist outside of this `#if MIN_VERSION_containers(0,5,8) ... #endif` block? The `traverseBia/Tree` rewrite rule is not declared within CPP, so it appears to be mismatched.

Reply:
Oops...